massa-execution/src/vm.rs

```diff
@@ -72,15 +75,15 @@ impl VM {
     ///
     /// # Parameters
     ///   * step: execution step to run
-    pub(crate) fn run_final_step(&mut self, step: &ExecutionStep) {
+    pub(crate) fn run_final_step(&mut self, step: ExecutionStep) {
         // check if that step was already executed as the earliest active step
-        let history_item = if let Some(cached) = self.pop_cached_step(step) {
+        let history_item = if let Some(cached) = self.pop_cached_step(&step) {
             // if so, pop it
             cached
         } else {
             // otherwise, clear step history an run it again explicitly
             self.step_history.clear();
-            self.run_step_internal(step)
+            self.run_step_internal(&step)
```

Comment:
I guess that you can reset these lines and just borrow the step

Reply:
I removed the borrow because it required borrowing in the run loop, which is weird because you actually want to take ownership of the values(for example in the new request to execute read-only stuff).